modules/openy_gc_shared_content/src/Controller/SharedContentController.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\openy_gc_shared_content\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Datetime\DateFormatter;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\media\Plugin\media\Source\Image;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * The SharedContentController class.
+ */
+class SharedContentController extends ControllerBase {
+
+  /**
+   * The date formatter.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatter
+   */
+  protected $dateFormatter;
+
+  /**
+   * The serializer.
+   *
+   * @var \Symfony\Component\Serializer\Serializer
+   */
+  protected $serializer;
+
+  /**
+   * SharedContentController constructor.
+   *
+   * @param \Drupal\Core\Datetime\DateFormatter $date_formatter
+   *   The date formatter.
+   * @param \Symfony\Component\Serializer\Serializer $serializer
+   *   The serializer.
+   */
+  public function __construct(DateFormatter $date_formatter, Serializer $serializer) {
+    $this->dateFormatter = $date_formatter;
+    $this->serializer = $serializer;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('date.formatter'),
+      $container->get('serializer')
+    );
+  }
+
+  /**
+   * Return the data.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   Return the JSON data.
+   */
+  public function index(string $type) {
+    if (!in_array($type, ['gc_video', 'vy_blog_post'])) {
+      return new JsonResponse(['error' => ['code' => '400']], 400);
+    }
+
+    return new JsonResponse(
+      $this->getData($type) + [
+        'method' => 'GET',
+        'status' => 200,
+      ]);
+  }
+
+  /**
+   * Get the data to be returned.
+   *
+   * @return array
+   *   The data to be returned.
+   */
+  public function getData($type) {
+    $node_storage = $this->entityTypeManager()->getStorage('node');
+    $serializer = $this->serializer;
+
+    $result = [];
+    $included = $included_loaded = [];
+    $meta = [];
+    $access_denied = FALSE;
+
+    // Fetch all nodes of $type that are published and shared.
+    $query = $node_storage->getQuery()
+      ->condition('type', $type)
+      ->condition('field_gc_share', 1)
+      ->condition('status', 1)
+      ->sort('created', 'DESC')
+      ->accessCheck(TRUE);
+    $nodes_ids = $query->execute();
+
+    if ($nodes_ids) {
+      foreach ($nodes_ids as $node_id) {
+        /** @var Drupal\node\Entity\Node $node */
+        $node = $node_storage->load($node_id);
```

Comment:
We could use static class.
Example: 
```
use Drupal\node\Entity\Node;
$node = Node::load($nid);
```
+
It is better to execute one `->loadMultiple` instead of loading one by one in a loop.
We could gather ids in a loop first and execute loadMultiple after that.

Reply:
I believe 202bcef64c1cea903dbe390b4821d8ca7038dda6 should resolve this. Thanks for the suggestion.